### PR TITLE
do not allow mapping to the same workload; remove important_knob_num

### PR DESCRIPTION
**Pull request by yangdsh**
I use clean_knob_data() in the periodic_task to filter out non-tunable knob before ranking. So I move clean_knob_data() into util.py.

I tested with having one workload in a project and having two workloads in a project. They behave as expected.

### CODE REVIEW

File: server/website/website/tasks/async_tasks.py [hunk 1]
```diff
@@ -1172,11 +1091,11 @@ def map_workload(map_workload_input):
             'rowlabels': rowlabels,
         }
 
-    if len(workload_data) == 0:
+    if len(workload_data) < 2:
```
Comment (dvanaken) [inline server/website/website/tasks/async_tasks.py:1094]: I don't think this is correct - if there are 2+ workloads then the workload data for the current session will still be added to `workload_data` and this check does not catch that. You need to explicitly check if the unique workload is the workload for this session and skip processing it if it is.

File: server/website/website/tasks/async_tasks.py [hunk 2]
```diff
@@ -1206,6 +1125,9 @@ def map_workload(map_workload_input):
 
     scores = {}
     for workload_id, workload_entry in list(workload_data.items()):
+        LOG.info('%s: %s', newest_result.workload.pk, workload_id)
+        if newest_result.workload.pk == workload_id:
+            continue
```
Comment (yangdsh) [inline server/website/website/tasks/async_tasks.py:1130]: I check the same workload here.

File: server/website/website/tasks/async_tasks.py (hunk 1 above)
Comment (yangdsh) [reply server/website/website/tasks/async_tasks.py:1094]: This is just a first-step filter. I check the workload afterward. Please see the below comment.

File: server/website/website/tasks/async_tasks.py (hunk 2 above)
Comment (dvanaken) [reply server/website/website/tasks/async_tasks.py:1130]: I see. But why not check if it's the same workload above when looping through the unique workloads? That way you don't spend time processing the same workload data that will never be used.

File: server/website/website/tasks/async_tasks.py (hunk 1 above)
Comment (yangdsh) [reply server/website/website/tasks/async_tasks.py:1094]: If there is more than 1 workload, it is guaranteed that there is a mapped workload, so we do not need to return early here.

File: server/website/website/tasks/async_tasks.py (hunk 2 above)
Comment (yangdsh) [reply server/website/website/tasks/async_tasks.py]: You are right.

**Pull request merged**